docs: add keptn certificate manager infos (#652)
keptn/lifecycle-toolkit: docs/content/en/docs/concepts/overview/klc-cert-manager/_index.md

---
title: Keptn Certificate Manager
icon: concepts
layout: quickstart
weight: 5
hidechildren: true # this flag hides all sub-pages in the sidebar-multicard.html
---

### Keptn Cert Manager

The Lifecycle Toolkit includes a Mutating Webhook which requires TLS certificates to be mounted as a volume in its pod. In version 0.6.0 and later, the certificate creation
is handled automatically by the [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md).

> The certificate is created as a secret in the `keptn-lifecycle-toolkit-system` namespace with a renewal threshold of 12 hours. If it expires, the [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md) renews it. The Lifecycle Toolkit operator waits for a valid certificate to be ready.

Reviewer: ```suggestion
The certificate is created as a secret in the `keptn-lifecycle-toolkit-system` namespace with a renewal threshold of 12 hours. 
If it expires, the [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md) renews it. 
The Lifecycle Toolkit operator waits for a valid certificate to be ready.
```

The certificate is mounted on an empty dir volume in the operator.

> When a certificate is left over from an older version, the webhook or the operator may generate errors because of an invalid certificate. ITo solve this, t delete the certificate and restart the operator.

Reviewer: ```suggestion
When a certificate is left over from an older version, the webhook or the operator may generate errors because of an invalid certificate. To solve this, delete the certificate and restart the operator.
```

---
title: How it works
icon: concepts
layout: quickstart
weight: 5
hidechildren: true # this flag hides all sub-pages in the sidebar-multicard.html
---

### Keptn Cert Manager

> The Lifecycle Toolkit includes a Mutating Webhook which requires TLS certificates to be mounted as a volume in its pod. Since version 0.6.0 the certificate creation

Reviewer: ```suggestion
The Lifecycle Toolkit includes a Mutating Webhook which requires TLS certificates to be mounted as a volume in its pod. Since version 0.6.0, the certificate creation
```

> The Lifecycle Toolkit includes a Mutating Webhook which requires TLS certificates to be mounted as a volume in its pod. Since version 0.6.0, the certificate creation

Reviewer 2: ```suggestion
The Lifecycle Toolkit includes a Mutating Webhook which requires TLS certificates to be mounted as a volume in its pod. In version 0.6.0 and later, the certificate creation
```

> is handled automatically by [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md).

Reviewer 2: ```suggestion
is handled automatically by the [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md).
```

> The certificate is created as a secret in the keptn-lifecycle-toolkit-system namespace with a renewal threshold of 12 hours. The Lifecycle Toolkit operator will be waiting for a valid certificate to be ready.

Reviewer: ```suggestion
The certificate is created as a secret in the `keptn-lifecycle-toolkit-system` namespace with a renewal threshold of 12 hours. In case of expiration, [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md) takes care of renewing it. The Lifecycle Toolkit operator will wait for a valid certificate to be ready.
```

> The certificate is created as a secret in the `keptn-lifecycle-toolkit-system` namespace with a renewal threshold of 12 hours. In case of expiration, [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md) takes care of renewing it. The Lifecycle Toolkit operator will wait for a valid certificate to be ready.

Reviewer 2: ```suggestion
The certificate is created as a secret in the `keptn-lifecycle-toolkit-system` namespace with a renewal threshold of 12 hours. If it expires, the [klt-cert-manager](https://github.com/keptn/lifecycle-toolkit/blob/main/klt-cert-manager/README.md) renews it. The Lifecycle Toolkit operator waits for a valid certificate to be ready.
```

The certificate will be mounted on an empty dir volume in the operator.

#### FAQ

> In case of certificates left over from a older version it could happen that the webhook or the operator errors due to invalid certificate. To solve this it is enough to delete de certificate and restart the operator.

Reviewer: ```suggestion
In case of certificates left over from an older version, it could happen that the webhook or the operator errors due to an invalid certificate. It is enough to delete the certificate and restart the operator to solve this.
```

Reviewer: could you please add the `kubectl` command to delete the secret and rollout-restart the pod?

> #### FAQ
> In case of certificates left over from an older version, it could happen that the webhook or the operator errors due to an invalid certificate. It is enough to delete the certificate and restart the operator to solve this.

Reviewer 2: ```suggestion
When a certificate is left over from an older version, the webhook or the operator may generate errors because of an invalid certificate. ITo solve this, t delete the certificate and restart the operator.
```
